src/map/script.cpp

```diff
@@ -25686,6 +25722,221 @@ BUILDIN_FUNC( openstylist ){
+	getitemdata_sub( SID_OPTIONID3, item_tmp->option[2].id );
+	getitemdata_sub( SID_OPTIONID4, item_tmp->option[3].id );
+	getitemdata_sub( SID_OPTIONID5, item_tmp->option[4].id );
+	getitemdata_sub( SID_OPTIONVALUE1, item_tmp->option[0].value );
+	getitemdata_sub( SID_OPTIONVALUE2, item_tmp->option[1].value );
+	getitemdata_sub( SID_OPTIONVALUE3, item_tmp->option[2].value );
+	getitemdata_sub( SID_OPTIONVALUE4, item_tmp->option[3].value );
+	getitemdata_sub( SID_OPTIONVALUE5, item_tmp->option[4].value );
+	getitemdata_sub( SID_OPTIONPARAM1, item_tmp->option[0].param );
+	getitemdata_sub( SID_OPTIONPARAM2, item_tmp->option[1].param );
+	getitemdata_sub( SID_OPTIONPARAM3, item_tmp->option[2].param );
+	getitemdata_sub( SID_OPTIONPARAM4, item_tmp->option[3].param );
+	getitemdata_sub( SID_OPTIONPARAM5, item_tmp->option[4].param );
+
+#undef getitemdata_sub
+
+	return SCRIPT_CMD_SUCCESS;
+}
+
+BUILDIN_FUNC(setitemdata)
+{
+	map_session_data* sd;
+
+	if (!script_rid2sd(sd)) {
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	int idx = script_getnum(st, 2);
+	if (idx < 0 || idx >= MAX_INVENTORY) {
+		ShowWarning("buildin_setitemdata: Index %d is out of the range 0-%d.\n", idx, MAX_INVENTORY - 1);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	if (sd->inventory_data[idx] == nullptr) {
+		ShowWarning("buildin_setitemdata: No item can be deleted from index %d of player %s (AID: %u, CID: %u).\n", idx, sd->status.name, sd->status.account_id, sd->status.char_id);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+	if (sd->inventory_data[idx]->type != IT_ARMOR && sd->inventory_data[idx]->type != IT_WEAPON && sd->inventory_data[idx]->type != IT_SHADOWGEAR) {
+		ShowError("buildin_setitemdata: Invalid type for item %u. The item must be equippable.\n", sd->inventory_data[idx]->nameid);
+		st->state = END;
+		return SCRIPT_CMD_FAILURE;
+	}
+
+	struct item *itm = &sd->inventory.u.items_inventory[idx];
+
+	switch( script_getnum(st, 3) ) {
+	case SID_ID: {
+		std::shared_ptr<item_data> id;
+
+		if (script_isstring(st, 4))
+			id = item_db.searchname( script_getstr(st, 4) );
+		else
+			id = item_db.find( script_getnum(st, 4) );
+
+		if (id == nullptr) {
+			ShowError("buildin_setitemdata: Nonexistant item '%s'.\n", script_getstr(st, 4));
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		if (id->type != IT_ARMOR && id->type != IT_WEAPON && id->type != IT_SHADOWGEAR) {
+			ShowError("buildin_setitemdata: Invalid type for item '%s'. The item must be equippable.\n", script_getstr(st, 4));
+			st->state = END;
+			return SCRIPT_CMD_FAILURE;
+		}
+		itm->nameid = id->nameid;
+		break;
+	}
+	case SID_IDENTIFY:
+		itm->identify = script_getnum(st, 4) > 0;
+		break;
+	case SID_REFINE:
+		itm->refine = cap_value( script_getnum(st, 4), 0, MAX_REFINE );
+		break;
+	case SID_ATTRIBUTE:
+		itm->attribute = script_getnum(st, 4) > 0;
+		break;
+	case SID_CARD1:
+		itm->card[0] = script_getnum(st, 4);
+		break;
```

Comment:
If the value to be set is not a special value it should have an item db check too.